File: results/py/COUNT_SUBARRAYS_EQUAL_NUMBER_1S_0S_1.cpp
int f_filled ( int arr [ ] , int n ) {
  map < int , int > mp ;
  int Sum = 0 ;
  int count = 0 ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    if ( ( arr [ i ] == 0 ) && ( arr [ i ] != - 1 ) ) arr [ i ] = - 1 ;
    Sum += arr [ i ] ;
    if ( ( Sum == 0 ) || ( Sum == 1 ) ) count ++ ;
    if ( ( Sum < mp . size ( ) ) || ( mp [ Sum ] == 0 ) ) count += mp [ Sum ] ;
    mp [ Sum ] = mp [ Sum ] + 1 ;
  }
  return count ;
}

File: results/java_codex/COUNT_SUBARRAYS_EQUAL_NUMBER_1S_0S_1.java
int countSubarrWithEqualZeroAndOne(int arr[], int n) {
        unordered_map<int, int> myMap;
        int sum = 0;
        int count = 0;
        for (int i = 0; i < n; i++) {
            if (arr[i] == 0) arr[i] = -1;
            sum += arr[i];
            if (sum == 0) count++;
            if (myMap.find(sum) != myMap.end()) count += myMap[sum];
            if (myMap.find(sum) == myMap.end()) myMap[sum] = 1;
            else myMap[sum] += 1;
        }
        return count;
    }